async-nats/src/lib.rs

```diff
@@ -493,7 +493,14 @@ impl Connector {
 
         for server_addr in &self.server_addrs {
             match self.try_connect_to(server_addr).await {
-                Ok(inner) => return Ok(inner),
+                Ok((info, connection)) => {
+                    for url in &info.connect_urls {
+                        self.server_addrs.push(url.parse()?);
```

Comment:
We should check for duplicates, as it will duplicate used address and further duplicate everything on each reconnect.

Reply:
sgtm, want to just check for contains or make it a set?

Reply:
Inserting to vec with checking for duplicates via contains sounds like hashSet with extra steps :).

Reply:
If I recall correctly, the sync client coutns the number of reconnects for each URL.
Would be good to checkout Go Client if it does the same. If yes- it should be a HashMap with key being URL and value - number of attempts.